samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadWriteTable.java

```diff
@@ -132,10 +152,29 @@ public void putAll(List<Entry<K, V>> entries) {
     List<Entry<K, V>> putRecords = records.stream()
         .filter(e -> e.getValue() != null).collect(Collectors.toList());
 
+    final long startNs = System.nanoTime();
+    CompletableFuture<Void> putFuture;
+    if (writeRateLimiter.isRateLimited()) {
+      putFuture = CompletableFuture
+          .runAsync(() -> writeRateLimiter.throttleRecords(putRecords), tableExecutor)
```

Comment:
two comments: a) not sure why we need two different APIs for writeRateLimiter.throttleRecords() and writeRateLimiter.throttle(key, value). If those two can be consolidated to a single common API, this putAll code section is almost the same as the write function you wrote above. b) if we can consolidate the writeRateLimiter function call, it seems that we can avoid duplicated code logic between single write function and the putAll function here, just passing in the IO method to be performed (i.e. writeFn.putAllAsync() or writeFn.putAsync()) and the metrics to record the latency (i.e. writeMetrics.putNs or writeMetrics.putAllNs).